.github/workflows/ci-cd-sharedkernel.yml

```diff
@@ -0,0 +1,118 @@
+name: SharedKernel CI/CD Pipeline
+
+on:
+  workflow_dispatch:
+  push:
+    paths: [ 'SharedKernel/src/**' ]
+  pull_request:
+    paths: [ 'SharedKernel/src/**' ]
+
+jobs:
+  ci:
+    name: Continuous Integration
+    runs-on: ubuntu-latest
+    outputs:
+      latest_version: ${{ steps.tag_generator.outputs.new_version }}
+    steps:
+      - name: Data gatherer
+        id: data_gatherer
+        shell: pwsh
+        run: |
+          # Get default branch
+          $repo = "${{ github.repository }}"
+          $defaultBranch = Invoke-RestMethod -Method GET -Uri https://api.github.com/repos/$repo | Select-Object -ExpandProperty default_branch
+          Write-Output "::set-output name=default_branch::$(echo $defaultBranch)"
+
+      - name: Setup .NET Core
+        id: setup_dotnet_core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.102
+
+      - name: Checkout repository
+        id: checkout_repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - if: steps.data_gatherer.outputs.default_branch == 'true'
+        name: Bump GH tag
+        id: tag_generator
+        uses: mathieudutour/github-tag-action@v5.4
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          default_bump: false
+          release_branches: ${{ steps.data_gatherer.outputs.default_branch }}
+
+      - name: Build solution
+        id: build_solution
+        shell: pwsh
+        run: |
+          dotnet build .\SharedKernel\SharedKernel.sln --configuration Release
+
+      - name: Run unit tests
+        id: run_unit_tests
+        shell: pwsh
+        run: |
+          dotnet test .\SharedKernel\SharedKernel.sln --configuration Release --no-build
+
+      - if: steps.tag_generator.outputs.new_version != ''
+        name: Create NuGet Package
+        id: create_nuget_package
+        shell: pwsh
+        run: |
+          nuget pack .\SharedKernel\src\PluralsightDdd.SharedKernel\PluralsightDdd.SharedKernel.csproj `
+          -OutputDirectory Artifacts `
+          -Properties Configuration=Release `
+          -Version ${{ steps.tag_generator.outputs.new_version }}
```

Comment:
Did you test the `-Version` parameter as a property in `-Properties` ? Did it work ?

Reply:
I tested the `-Version` parameter as a property in `-Properties` and it is not working.
As we discussed the NuGet CLI documentation is not updated.
the `-Version` parameter is working the same way like in the .nuspec file.

Reply:
Hmm, it should work.
What is the code that did work for you ?

Reply:
@aleks-ivanov My mistake the code is working and its working with `-Properties Version` I miss spelled the code. I will redo it and upload the changes right away.

Reply:
@aleks-ivanov Done !